### ISSUE TITLE
Unite/unify tensor.unfold and torch.nn.functional.unfold

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#44991]
Two versions exist for seemingly same functionality:

1) tensor.unfold supports Long tensors (F.unfold does not: https://github.com/pytorch/pytorch/issues/44989)
2) tensor.unfold returns different shape layout
3) tensor.unfold has differnet argument names: size/step instead of kernel_size/stride
4) tensor.unfold does not support padding (while F.unfold does)

**Comment by vadimkantorov** [comment]
Also, unfolding 1x1x300000 and then applying torch.mode is super slow, even on GPU. Maybe torch.mode doesn't batch on the good dimension...

**Comment by vadimkantorov** [comment]
Also tensor.unfold seems 1d-only and much more constrained compared to F.unfold

It would also help if the doc included explanation how "implicit padding" can work if only strides are modified and no copies is done (is it true?)?

**Comment by vadimkantorov** [comment]
Also, it seems that 2d unfold can also be implemented with stride tricks (mostly zero-copy)?

https://stackoverflow.com/questions/73013707/performing-im2col-with-numpy-lib-stride-tricks-as-strided

It also exists in NumPy, right? https://numpy.org/devdocs/reference/generated/numpy.lib.stride_tricks.sliding_window_view.html @rgommers

**Comment by rgommers** [comment]
Yes, it's zero-copy in numpy. 

The suggestions in this issue seem right to me - some cleanup and unification would make sense.

**Comment by vadimkantorov** [comment]
> Yes, it's zero-copy in numpy.

Btw, can it support some constant padding? In theory one could first F.pad the input tensor, and then unfold should even support padding by doing stride tricks on the padded tensor

**Comment by vadimkantorov** [comment]
Related issues:
- https://github.com/pytorch/pytorch/issues/44989
- https://github.com/pytorch/pytorch/issues/60466
